src/main/java/com/minecolonies/coremod/colony/buildings/AbstractSchematicProvider.java

```diff
@@ -377,20 +378,30 @@ protected void updateTEDataFromSchematic()
 
         try
         {
-            saveUpdateTEDataFromSchematic(te);
+            unsaveUpdateTEDataFromSchematic(te);
+            return;
         }
         catch (final Exception ex)
         {
             Log.getLogger().warn("TileEntity with invalid data, restoring correct data from schematic.");
             te.setSchematicName(this.getSchematicName() + Math.max(1, buildingLevel));
-            saveUpdateTEDataFromSchematic(te);
+        }
+
+        try
+        {
+            unsaveUpdateTEDataFromSchematic(te);
+        }
+        catch (final Exception ex)
+        {
+            Log.getLogger().warn("Building with invalid data, setting building as deconstructed.");
```

Comment:
Not true after the last edit. :)

Reply:
Leftover from int cast attempt =D

Reply:
No need to warn if we warn the players directly too.